### PR TITLE
Add doc mentioning the biggest differences of dev-mode vs the production application

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM

File: docs/src/main/asciidoc/dev-mode-differences.adoc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+=== Health UI
+
+The `quarkus-smallrye-health` extension will expose the Health UI when Quarkus is running in dev-mode. link:smallrye-health.adoc#ui[This] section provides additional information.
+
+=== Mock mailer
+
+The `quarkus-mailer` extension will enable an in-memory mock mail server when Quarkus is running in dev-mode. See link:mailer.adoc#testing[this] for more details.
+
+== Performance implications
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-mode-differences.adoc:76]:
```suggestion

=== gRPC Reflection Service

The gRPC Reflection Service is enabled in dev mode by default. That lets you use tools such as `grpcurl`. In production mode, the reflection service is disabled. You can enable it explicitly using `quarkus.grpc-server. enable-reflection-service=true`

== Performance implications
```

**Review by cescoffier — DISMISSED** [review]

File: docs/src/main/asciidoc/dev-mode-differences.adoc (hunk 1 above)
Comment (gastaldi) [reply docs/src/main/asciidoc/dev-mode-differences.adoc:76]: Is the extra space in the property name intentional or a typo? I think it's a typo?

**Comment by michalszynkiewicz** [comment]
Not sure if it's worth mentioning here but `quarkus.grpc.server.instances` has no effect in the dev mode (Number of gRPC server verticle instances.)

File: docs/src/main/asciidoc/dev-mode-differences.adoc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+In dev-mode minimizing the runtime footprint of the application is not the primary objective (although Quarkus still starts plenty fast and consumes little memory even in dev-mode) - the primary objective
+is enabling developer joy.
+Therefore, many more classes are loaded and build time operations also take place every time a live-reload is performed.
+
+In contrast, in a production application the main objective for Quarkus is to consume the least amount of memory and startup in the smallest amount of time.
+Thus, when running the production application, build time operations are not performed (by definition) and various infrastructure classes needed at build time at not present at all at runtime.
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/dev-mode-differences.adoc:83]:
```suggestion
Thus, when running the production application, build time operations are not performed (by definition) and various infrastructure classes needed at build time are not present at all at runtime.
```

**Comment by geoand** [comment]
All comments were addressed

File: docs/src/main/asciidoc/dev-mode-differences.adoc [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+The `quarkus-mailer` extension will enable an in-memory mock mail server when Quarkus is running in dev-mode. See link:mailer.adoc#testing[this] for more details.
+
+
+=== gRPC
+
+* The gRPC Reflection Service is enabled in dev mode by default. That lets you use tools such as `grpcurl`. In production mode, the reflection service is disabled. You can enable it explicitly using `quarkus.grpc-server.enable-reflection-service=true`.
+
+* In dev-mode, `quarkus.grpc.server.instances` has no effect.
+
+
```
Comment (maxandersen) [inline docs/src/main/asciidoc/dev-mode-differences.adoc:83]: maybe add note/mention that there could be more - depends on each extension added to the project how much they support.

Comment (geoand) [reply docs/src/main/asciidoc/dev-mode-differences.adoc:83]: Done

**Review by cescoffier — APPROVED** [review]

**Comment by geoand** [comment]
@gsmet is there anyway we can get this into the website for `1.13`?

**Comment by gsmet** [comment]
@geoand is it urgent or can it wait for 1.13.1.Final? I usually release it one week after the 0.Final.

**Comment by geoand** [comment]
It can certainly wait

**Comment by gsmet** [comment]
OK, I marked it for inclusion in 1.13.1.Final.

**Comment by geoand** [comment]
Thanks 👍

**Pull request merged**